src/goto-harness/recursive_initialization.cpp

```diff
@@ -122,6 +122,40 @@ void recursive_initializationt::initialize(
   {
     initialize_nondet(lhs, depth, known_tags, body);
 
+code_blockt recursive_initializationt::build_constructor_body(
+  const exprt &depth_symbol,
+  const exprt &result_symbol,
+  const optionalt<exprt> &size_symbol,
+  const optionalt<irep_idt> &lhs_name)
+{
+  PRECONDITION(result_symbol.type().id() == ID_pointer);
+  const typet &type = result_symbol.type().subtype();
+  if(type.id() == ID_struct_tag)
+  {
+    return build_struct_constructor(depth_symbol, result_symbol);
+  }
+  else if(type.id() == ID_pointer)
+  {
+    if(lhs_name.has_value())
+    {
+      if(should_be_treated_as_cstring(*lhs_name) && type == char_type())
```

Comment:
Just a question, since I'm not sure why it's happening: 

Why in other places we compare the `type.id()`, like for instance in line 137 `type.id() == ID_pointer` and here we compare the type itself with the constructor result? Are we sure equality is working as it should? If yes, should all of these be harmonised?

Reply:
That's just my being lazy and not finding out what type-id `char` should have.